### DISCUSSION TITLE
Textarea, x-model and line breaks

### DISCUSSION THREAD

**Discussion by mattgreenfield** [alpinejs/alpine#2358] [1. Help] [answered]
Is there a way to preserve line breaks created by textareas?

Here's a quick demo, if you add a line break (hit return) in the textarea, it's not displayed in the rendered text.
https://codepen.io/gmattgreenfield/pen/KKvBENz

I appreciate this isn't always desired so maybe it's optional?

Thanks

**Comment by danddanddand** [comment]
Switch out your x-text with something like this `x-html="title.replaceAll('\n', '<br>')"`

**Reply by mattgreenfield** [reply]
Thank you for your suggestion. Although it appears to work it would allow users to write HTML in the text input, that seems dangerous!

**Reply by rohait0717** [reply]
> Switch out your x-text with something like this `x-html="title.replaceAll('\n', '<br>')"`

Great that work for me !!! Thanks alot

**Comment by SimoTod** [comment, marked as answer]
Alternatively use a `<pre>` tag

**Reply by mattgreenfield** [reply]
Thank you, I think a `<pre>` tag, with some reset CSS will work.

**Comment by nikolaysm** [comment]
This can also be achieved by adding CSS `white-space` rule:

```
<div style="white-space: pre;" x-text="title"></div>
```
